docs/using_eessi/building_on_eessi.md

```diff
@@ -6,67 +6,109 @@ Building on top of EESSI with EasyBuild is relatively straightforward. One cruci
 ### Starting the EESSI software environment
 Start your environment as described [here](../using_eessi/setting_up_environment.md)
 
-### Configure EasyBuild
-To configure EasyBuild, first, check out the [EESSI software-layer repository](https://github.com/EESSI/software-layer.git). We advise you to check out the branch corresponding to the version of EESSI you would like to use.
+### Using the EESSI-extend module
+The `EESSI-extend` module facilitates building on top of EESSI using EasyBuild. It does a few key things:
 
-If you are unsure which version you are using, you can run
+1. It configures EasyBuild to match how the rest of the EESSI software is built
+2. It configures EasyBuild to use a certain installation path (e.g. in your homedir), taking into account the hardware architecture you are building on
+3. It adds the relevant subdirectory from your installation path to your `MODULEPATH`, to make sure your newly installed modules are available
+4. It loads the EasyBuild module
+
+The `EESSI-extend` module recognizes a few environment variables. To print an up-to-date list, check the module itself
 ```
-echo ${EESSI_VERSION}
+module help EESSI-extend/2023.06-easybuild
 ```
-to check it.
 
-To build on top of e.g. version `2023.06` of the EESSI software stack, we check it out, and go into that directory:
+The installation prefix is determined by `EESSI-extend` through the following logic:
+1. If `EESSI_CVMFS_INSTALL` is set, software inst installed in `EESSI_SOFTWARE_PATH`. This variable shouldn't be used by users and would only be used by CVMFS administrators of the EESSI repository.
+2. If `EESSI_SITE_INSTALL` is set (and this is set by default when the EESSI environment is initialized) and the EESSI site installation prefix (`/cvmfs/software.eessi.io/host_injections/...`) is writeable, this prefix will be used. This is typically where sites hosting a system that has EESSI deployed would install additional software on top of EESSI and make it available to all their users.
```

Comment:
```suggestion
2. If `EESSI_SITE_INSTALL` is set and the EESSI site installation prefix (`/cvmfs/software.eessi.io/host_injections/...`) is writeable, this prefix will be used. This is typically where sites hosting a system that has EESSI deployed would install additional software on top of EESSI and make it available to all their users.
```
This is not set by default, what is done by default is _exposing_ the modules that are installed with this approach.

Reply:
Ah, I figured that was also a `PATH`-like variable. It isn't. Looking at the `EESSI-extend` module:

```
elseif (os.getenv("EESSI_SITE_INSTALL") ~= nil) then
  -- Make sure no other EESSI install environment variables are set
  if ((os.getenv("EESSI_PROJECT_INSTALL") ~= nil) or (os.getenv("EESSI_USER_INSTALL") ~= nil)) then
    LmodError("You cannot use EESSI_SITE_INSTALL in combination with any other EESSI_*_INSTALL environment variables")
  end
  easybuild_installpath = string.gsub(os.getenv("EESSI_SOFTWARE_PATH"), 'versions', 'host_injections')
```

Not really a docs question, but it's a bit awkward that this path is recalculated when the EESSI modulefile already sets `EESSI_SITE_MODULEPATH`. Admittedly, that is _not_ the installpath (that would be 2 dirs up).

Ideally, I would _also_ set `EESSI_SITE_INSTALLPATH` in the EESSI modulefile (and make `EESSI_SITE_MODULEPATH relative to that). But that's a different story.

The point is, right now, it's very hard to explain to a user _where_ this site installation will end up. We don't have an environment variable pointing to that prefix :\ For now, that `...` is the best I can do, but it's very a-specific.